tests/test_util_search.py

```diff
@@ -172,12 +172,12 @@ def test_search_return_list_globbed():
         }
     res = dpath.util.search(tdict, '/a/b/[02]')
     assert(isinstance(res['a']['b'], list))
-    assert(len(res['a']['b']) == 2)
-    assert(res['a']['b'] == [0, 2])
+    assert(len(res['a']['b']) == 3)
+    assert(res['a']['b'] == [0, None, 2])
```

Comment:
`util.search(x, '/a/b/[02]')` => This to me means: Search through x for paths that match /a/b/0 and /a/b/2 and return a view of the result. Since x['a']['b'] is a sequence we have to fill in the missing elements with something or else the view is not self consistent (and doesn't satisfy the doc string requirement that it should contain the keys that matched the glob).

By self consistent I mean that if I do a search I should be able to also do a search on that search and get the same result:

```
v1 = util.search(x, '/a/b/[02]')
v2 = util.search(v1, '/a/b/[02]')
assert(v1 == v2)
```

It doesn't make sense to me to have a search that returns a resulting view that then wouldn't match the original search.

It may be that we need either a different function or a flag to control the behavior when creating sparse arrays.